cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/namespace/NamespaceServiceTest.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.runtime.namespace;
+
+import co.cask.cdap.common.exception.AlreadyExistsException;
+import co.cask.cdap.common.exception.NotFoundException;
+import co.cask.cdap.internal.app.namespace.NamespaceService;
+import co.cask.cdap.internal.app.services.http.AppFabricTestBase;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.NamespaceMeta;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Tests for {@link NamespaceService}
+ */
+public class NamespaceServiceTest extends AppFabricTestBase {
+  private static final NamespaceService namespaceService = getInjector().getInstance(NamespaceService.class);
```

Comment:
Hmmm, I think it could be possible. Don't think there is a good enough way to retrieve the actual service started in the ``AppFabricTestBase``. However, I do not think there should be multiple instances of ``NamespaceService`` to begin with, no? Ensured that by making ``NamespaceService`` a ``@Singleton``.